_examples/gorgonia_iris/main.go

```diff
@@ -0,0 +1,186 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"math"
+	"os"
+
+	"github.com/c-bata/goptuna"
+	"github.com/c-bata/goptuna/rdb"
+	"github.com/c-bata/goptuna/tpe"
+	"github.com/go-gota/gota/dataframe"
+	"github.com/go-gota/gota/series"
+	"github.com/jinzhu/gorm"
+	"gonum.org/v1/gonum/mat"
+	"gorgonia.org/gorgonia"
+	"gorgonia.org/tensor"
+
+	_ "github.com/jinzhu/gorm/dialects/sqlite"
+)
+
+var (
+	dataset string
+)
+
+func init() {
+	flag.StringVar(&dataset, "dataset", "iris.csv", "File path to iris dataset")
+	flag.Parse()
+}
+
+func main() {
+	db, err := gorm.Open("sqlite3", "db.sqlite3")
+	if err != nil {
+		log.Fatal("failed to open db:", err)
+	}
+	rdb.RunAutoMigrate(db)
+	storage := rdb.NewStorage(db)
+
+	study, err := goptuna.CreateStudy(
+		"gorgonia-iris",
+		goptuna.StudyOptionStorage(storage),
+		goptuna.StudyOptionSampler(tpe.NewSampler()),
+		goptuna.StudyOptionSetDirection(goptuna.StudyDirectionMaximize),
+	)
+	if err != nil {
+		log.Fatal("failed to create study: ", err)
+	}
+	err = study.Optimize(objective, 50)
+	if err != nil {
+		log.Fatal("failed to create study: ", err)
+	}
+
+	v, _ := study.GetBestValue()
+	params, _ := study.GetBestParams()
+	log.Printf("Best evaluation=%f (learning_rate=%f)",
+		v, params["learning_rate"].(float64))
+}
+
+// https://www.kaggle.com/amarpandey/implementing-linear-regression-on-iris-dataset/notebook
+func objective(trial goptuna.Trial) (float64, error) {
+	g := gorgonia.NewGraph()
+	x, y := getXYMat()
+	xT := tensor.FromMat64(mat.DenseCopyOf(x))
+	yT := tensor.FromMat64(mat.DenseCopyOf(y))
+
+	s := yT.Shape()
+	yT.Reshape(s[0])
+
+	X := gorgonia.NodeFromAny(g, xT, gorgonia.WithName("x"))
+	Y := gorgonia.NodeFromAny(g, yT, gorgonia.WithName("y"))
+	theta := gorgonia.NewVector(
+		g,
+		gorgonia.Float64,
+		gorgonia.WithName("theta"),
+		gorgonia.WithShape(xT.Shape()[1]),
+		gorgonia.WithInit(gorgonia.Gaussian(0, 1)))
+
+	pred, err := gorgonia.Mul(X, theta)
+	if err != nil {
+		return 0, err
+	}
+
+	// Gorgonia might delete values from nodes so we are going to save it
+	// and print it out later
+	var predicted gorgonia.Value
+	gorgonia.Read(pred, &predicted)
+
+	predError, err := gorgonia.Sub(pred, Y)
+	if err != nil {
+		return 0, err
+	}
+	squaredError, err := gorgonia.Square(predError)
+	if err != nil {
+		return 0, err
+	}
+	cost, err := gorgonia.Mean(squaredError)
+	if err != nil {
+		return 0, err
+	}
+
+	if _, err := gorgonia.Grad(cost, theta); err != nil {
+		return 0, err
+	}
+
+	machine := gorgonia.NewTapeMachine(g, gorgonia.BindDualValues(theta))
+	defer machine.Close()
+
+	model := []gorgonia.ValueGrad{theta}
+	learnRate, _ := trial.SuggestLogFloat("learning_rate", 1e-5, 1e-1)
+	solver := gorgonia.NewVanillaSolver(gorgonia.WithLearnRate(learnRate))
```

Comment:
Optimize learning rate.

Reply:
The following suggestion is more interesting as an Goptuna example.

```go
	var solver gorgonia.Solver
	solverName, _ := trial.SuggestCategorical("solver", []string{"Adam", "Vanilla"})
	if solverName == "Adam" {
		solver = gorgonia.NewAdamSolver()
	} else if solverName == "Vanilla" {
		learnRate, _ := trial.SuggestLogFloat("vanilla_learning_rate", 1e-5, 1e-1)
		solver = gorgonia.NewVanillaSolver(gorgonia.WithLearnRate(learnRate))
	}
```

I created a pull request: https://github.com/c-bata/goptuna/pull/119